code/game/mecha/mech_fabricator.dm

```diff
@@ -502,8 +502,9 @@
 	for(var/v in stored_research.researched_designs)
 		var/datum/design/D = SSresearch.techweb_design_by_id(v)
 		if(D.build_type & MECHFAB)
-			if(ispath(D.build_path, /obj/item/mecha_parts/mecha_equipment/weapon) && !combat_parts_allowed)
-				continue
+			if(ispath(D.build_path, /obj/item/mecha_parts/mecha_equipment/weapon) && !combat_parts_allowed) // Yogs -- ID swiping for combat parts
```

Comment:
Why not make it a variable such as `var/combat_design = FALSE` then set it for the `weapon` type and make any specific changes from there

Reply:
It can become its own variable if there ends up being any more than one example; this is coded as a snowflake patch because there is honestly only this one example I know of.